### PR TITLE
enable the author to fetch a single drafted article

### PR DESCRIPTION
**Pull request by truestbyheart**
#### What does this PR do?
* It enables the author to fetch a single drafted article.
#### Description of Task to be completed?
 * Create end-point for fetching a single drafted article.
#### How should this be manually tested?
* Clone the repo to the local machine.
* Run ``npm install``   to install the dependencies.
* Set up the   ``.env`` as instructed by the   ``.env-example``
* Run  ``npm setup:dev``.
*Run ``npm start``.
* Send a GET request to ``/api/articles/drafts/:slug``.
#### What are the relevant pivotal tracker stories?
#[168242065](https://www.pivotaltracker.com/story/show/168242065)

### CODE REVIEW

**Review by nkpremices — APPROVED** [review]
Good job @truestbyheart

**Review by MCFrank16 — APPROVED** [review]

File: src/api/controllers/articleController.js [hunk 1]
```diff
@@ -232,8 +232,9 @@ export default class ArticleController {
       description: description || req.Existing.description,
       body: body || req.Existing.body,
       category: category || req.Existing.category,
-      coverImage: coverImage || req.Existing.coverImage
+      coverImage
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @truestbyheart why did you edit this line?

File: tests/updateArticle.test.js [hunk 2]
```diff
@@ -121,7 +122,8 @@ describe('testing for the article update controller', () => {
       .set('authorization', UserToken)
       .send({
         body: '',
-        tags: ['Laravel', 'php', 'IOT', 'iot2']
+        tags: ['Laravel', 'php', 'IOT', 'iot2'],
+        coverImage: 'http://localhost:3000/ver.jpg'
```
Comment (MemunaHaruna) [inline tests/updateArticle.test.js:126]: @truestbyheart only the author of a draft article should be able to view/update it. Is there a check for this?

Comment (truestbyheart) [reply tests/updateArticle.test.js:126]: I didn't implement that let work on it.

File: src/api/controllers/articleController.js (hunk 1 above)
Comment (truestbyheart) [reply src/api/controllers/articleController.js]: Oooh... I was trying to troubleshoot why the image was not being uploaded so I forgot to bring it back let me put it back

File: tests/article.test.js [hunk 3]
```diff
@@ -141,3 +141,30 @@ describe('create  article', () => {
       });
   });
 });
+
+describe('GET /drafts/:slug', () => {
```
Comment (MemunaHaruna) [inline tests/article.test.js:146]: @truestbyheart can you quickly add a test that shows that when a user who isn't the author requests for a draft article, it returns an error?

**Review by MemunaHaruna — APPROVED** [review]

**Pull request merged**